wallet/wallet.c

```diff
@@ -185,7 +185,8 @@ const struct utxo **wallet_select_coins(const tal_t *ctx, struct wallet *w,
 		db_rollback_transaction(w->db);
```

Comment:
And if "should_commit" is true here?

Reply:
Yep, fixed it. We would probably have caught it the next time we tried to start something with `transactional_msg_handler` which checks for leaked transactions. We could also do something similar to the `tmpctx` check in `debug_poll` for transactions, but I wanted to keep it simple at first.